### PR TITLE
Add support for manual builds, git source for webhook/pubsub triggered builds and filter field

### PR DESCRIPTION
**Pull request by abhinavrau**
Fixes hashicorp/terraform-provider-google#10779
Fixes hashicorp/terraform-provider-google#10434
Fixes hashicorp/terraform-provider-google#10387
Fixes hashicorp/terraform-provider-google#9755
Fixes hashicorp/terraform-provider-google#8374
Fixes hashicorp/terraform-provider-google#9189
Fixes hashicorp/terraform-provider-google#9883

Add support for REST fields `sourceToBuild` and `gitFileSource` which adds support for manual builds as well as adds support for webhook and pubsub triggered builds with source code. 
Also added support for filter field that is used with webhook and pubsub triggered builds.
Also added examples for webhook, pubsub (with filter) and manual build.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudbuild: Add support for manual builds, git source for webhook/pubsub triggered builds and filter field
```

### CODE REVIEW

File: mmv1/products/cloudbuild/api.yaml [hunk 1]
```diff
@@ -83,8 +83,83 @@ objects:
...
+          - github
+          - pubsub_config
+          - webhook_config
+          - source_to_build
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'uri'
+            required: true
+            description: |
+              The URI of the repo (optional). If unspecified, the repo from which
```
Comment (megan07) [inline mmv1/products/cloudbuild/api.yaml:147]: Is this the correct description? It looks like it's marked required, but described as optional.

**Review by megan07 — CHANGES_REQUESTED** [review]
Other than that description change, I think everything else looks good! Thank you!

File: mmv1/products/cloudbuild/api.yaml (hunk 1 above)
Comment (abhinavrau) [reply mmv1/products/cloudbuild/api.yaml]: Thanks for reviewing @megan07. I have updated the description.

**Review by megan07 — APPROVED** [review]
Looks great! Thanks so much!

**Comment by wyardley** [comment]
Does this support the manual _approval_ parameter for builds as well? I had assumed so, but it looks like this is relating to manually _triggering_ builds, not manually approving builds?

**Comment by slevenick** [comment]
What do you mean by manually approving builds? Is there a particular field/method in the REST API you are looking to use?

**Comment by abhinavrau** [comment]
This supports creating builds to be manually invoked. This does not support
approval

**Comment by wyardley** [comment]
@slevenick @abhinavrau yeah, was looking for https://github.com/hashicorp/terraform-provider-google/issues/9880 tho after I read the magic modules PR, realized this was the fix for something different.

**Comment by abhinavrau** [comment]
@slevenick I have submitted a PR #5837 for https://github.com/hashicorp/terraform-provider-google/issues/9880. Let me know how it works out.

**Pull request merged**